jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/component/Environment.java

```diff
@@ -13,41 +13,68 @@
 
 package org.eclipse.jetty.util.component;
 
-import java.io.IOException;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.Map;
-import java.util.TreeMap;
 
 import org.eclipse.jetty.util.Attributes;
-import org.eclipse.jetty.util.TypeUtil;
 
 /**
  * A named runtime environment containing a {@link ClassLoader} and {@link Attributes}.
  */
 public interface Environment extends Attributes
 {
     // Ensure there is a core environment for possible later deployments to it
-    Environment CORE = ensure("core");
+    Environment CORE = ensure("core", Environment.class.getClassLoader());
 
+    /**
+     * Gets all existing environments.
+     * @return the environments
+     */
     static Collection<Environment> getAll()
     {
-        return Collections.unmodifiableCollection(Named.__environments.values());
+        return Collections.unmodifiableCollection(NamedEnvironment.ENVIRONMENTS.values());
     }
-    
+
+    /**
+     * Gets the environment with the given name.
+     * @param name the environment name
+     * @return the environment, or null if no environment with such name exists
+     */
     static Environment get(String name)
     {
-        return Named.__environments.get(name);
+        return NamedEnvironment.ENVIRONMENTS.get(name);
     }
 
-    static Environment ensure(String name)
+    /**
+     * Gets the environment with the given name, creating it with the default classloader if necessary.
+     *
+     * @param name the environment name
+     * @return the environment
+     * @throws IllegalStateException if an environment with the given name but a non-default classloader already exists
```

Comment:
There is no concept of default classloader anymore.

Reply:
updated javadoc